### PR TITLE
DOC: add details on Meson build debugging and introspection

### PR DESCRIPTION
**Pull request by rgommers**
[skip actions][skip azp]

### CODE REVIEW

File: doc/source/dev/contributor/meson_advanced.rst [hunk 1]
```diff
@@ -86,3 +86,228 @@ to run the ``scipy.linalg`` tests for builds with both compilers, do::
...
+
+Introspecting build steps
+=========================
+
+When you have an issue with a particular Python extension module or other build
+target, there are a number of ways to figure out what the build system is doing
+exactly. Beyond looking at the ``meson.build`` content for the target of
+interest, these include:
+
+1. Reading the generated ``build.ninja`` file in the build directory (available
```
Comment (matthew-brett) [inline doc/source/dev/contributor/meson_advanced.rst:98]: Did you mean to complete this sentence after "available"?

**Review by matthew-brett — COMMENTED** [review]
Clarification comment.

File: doc/source/dev/contributor/meson_advanced.rst (hunk 1 above)
Comment (rgommers) [reply doc/source/dev/contributor/meson_advanced.rst]: good catch, fixed

**Comment by rgommers** [comment]
@matthew-brett if you are happy with this, do you want to hit the green button?

**Pull request merged**